ogr/services/github/token_managers/tokman_manager.py

```diff
@@ -0,0 +1,43 @@
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+
+
+import requests
+
+
+import github
+
+
+from ogr.abstract import GithubTokenManager
+from ogr.exceptions import OgrException
+
+
+class TokmanGithubTokenManager(GithubTokenManager):
+    def __init__(self, instance_url: str):
+        self.__instance_url = instance_url
+
+        try:
+            # check if running correctly
+            health = requests.get(f"{self.__instance_url}/api/health")
+
+            if not health.ok or health.json().get("message", None) != "ok":
+                raise OgrException("Tokman instance is not running")
+        except OgrException as ex:
+            raise ex
+        except Exception as ex:
+            raise OgrException(f"Couldn't connect to Tokman instance: {ex}")
+
+    def get_instance(self, namespace: str, repo: str) -> github.Github:
+        response = requests.get(f"{self.__instance_url}/api/{namespace}/{repo}")
+
+        json_response = response.json()
+        token = json_response.get("access_token", None)
+
+        if not response.ok or not token:
+            # haven't managed to get a token
+            error = json_response.get("error", "no message provided")
+            raise OgrException(
+                f"Couldn't retrieve token from Tokman: {error}"
+            )
```

Comment:
Usually checking the response for errors is done *before* any information is retrieved from the response.

Additionally: any reason for not using `response.raise_for_status()`? Should make thing simpler.

Reply:
> Additionally: any reason for not using response.raise_for_status()? Should make thing simpler.

Also done, but it raises generic `BAD REQUEST` exception right now... It's in separate commit in case we would prefer raising `OgrException` with "prettier" message.

Reply:
Indeed, this will not show the error message, which is not nice.

The main issue with the previous solution is that it's using `response.json()` before checking the status codes. Although the tokman API will return JSON for 200 and 400, the response body for other status codes might not be valid JSON, in which case this call will fail.

How about not reading the response as a JSON in case of an error, but simply using whatever there is in `response.text`?

Reply:
Redone with `response.text`